cms/djangoapps/contentstore/views/course_optimizer.py

```diff
@@ -0,0 +1,212 @@
+"""
+These views handle all actions in Studio related to link checking of
+courses
+"""
+
+
+import json
+import logging
+import os
+from wsgiref.util import FileWrapper
+
+from django.conf import settings
+from django.contrib.auth.decorators import login_required
+from django.core.exceptions import PermissionDenied
+from django.db import transaction
+from django.http import Http404, HttpResponse, StreamingHttpResponse
+from django.utils.translation import gettext as _
+from django.views.decorators.csrf import ensure_csrf_cookie
+from django.views.decorators.http import require_GET, require_http_methods
+from opaque_keys.edx.keys import CourseKey
+from user_tasks.conf import settings as user_tasks_settings
+from user_tasks.models import UserTaskArtifact, UserTaskStatus
+
+from common.djangoapps.student.auth import has_course_author_access
+from common.djangoapps.util.json_request import JsonResponse
+from common.djangoapps.util.views import ensure_valid_course_key
+from cms.djangoapps.contentstore.xblock_storage_handlers.view_handlers import get_xblock
+from cms.djangoapps.contentstore.xblock_storage_handlers.xblock_helpers import usage_key_with_run
+from xmodule.modulestore.django import modulestore  # lint-amnesty, pylint: disable=wrong-import-order
+
+from ..tasks import CourseLinkCheckTask, check_broken_links
+from ..utils import reverse_course_url
+
+__all__ = [
+    'link_check_handler',
+    'link_check_status_handler',
+]
+
+log = logging.getLogger(__name__)
+
+# Tuple containing zero or more filters for UserTaskStatus listing REST API calls.
+STATUS_FILTERS = user_tasks_settings.USER_TASKS_STATUS_FILTERS
+
+
+@transaction.non_atomic_requests
+@ensure_csrf_cookie
+@login_required
+@require_http_methods(('POST'))
+@ensure_valid_course_key
+def link_check_handler(request, course_key_string):
+    """
+    The restful handler for checking broken links in a course.
+
+    POST
+        Start a Celery task to check broken links in the course
+
+    The Studio UI uses a POST request to start the link check asynchronously, with
+    a link appearing on the page once it's ready.
+    """
+    course_key = CourseKey.from_string(course_key_string)
+    if not has_course_author_access(request.user, course_key):
+        raise PermissionDenied()
+    courselike_block = modulestore().get_course(course_key)
```

Comment:
What's a `courselike_block`? Naming isn't that clear to me

Reply:
We prefer the term "learning context" over "courselike" now.